docs/shards.adoc

```diff
@@ -74,6 +74,7 @@ doesn't generate a conflict, thus generating a new _shard.lock_ file.
 --production:: same as _--frozen_ and _--without-development_
 --skip-postinstall:: Does not run postinstall of dependencies.
 --skip-executables:: Does not install executables.
+--parallel-fetch:: Number of git fetches to perform in parallel (default: 8).
```

Comment:
IIUC this works for every resolver (implementing `update_local_cache`), not just _git_?
If so, I'd rename it to `parallel-jobs` or sth like that.

Reply:
I thought `fetch` would make the most sense for all resolvers where this applies
since it's about the data-fetching and nothing else - each VCS might call that step
something different but ultimately it's a "fetch".

It's only implemented for git atm, that's why I mentioned that one in the description.

We could change it but to me `parallel-jobs` feels less to the point than `parallel-fetch`? 🤔

Reply:
> It's only implemented for git atm, that's why I mentioned that one in the description.

I don't follow. You're calling `Resolver#update_local_cache` for every resolver regardless, so how come you state it's implemented only for git?

Reply:
Heck... you are right.

I got confused because my first iteration was git-specific,
I totally forgot that it now indeed applies to all of them. 🙈 

Ouch, that means this PR probably [breaks hg](https://github.com/crystal-lang/shards/blob/f3a30698ccb71c0f796d4931413f35c518e3eeb0/src/resolvers/hg.cr#L434).
The hg-side will need a change similar to [this](https://github.com/crystal-lang/shards/pull/539/files#diff-da83517c6d5d940ffa16fa8f176bbe55ac9b5ef3abafe8336d35b48a6be171a1L414).

Sigh. This makes things more complicated.

I can port the change from git.cr to hg.cr - but I haven't wrapped my head
around the spec-suite enough to see how to make a test to validate it.

It looks like the current [integration test](https://github.com/crystal-lang/shards/blob/635d36af83b3adfbfbc8af88e5f00a3c5af5d585/spec/integration/install_spec.cr) only uses git URLs (no `hg_url()` in there).
That may be the reason why it didn't catch the breakage - or maybe
it just doesn't exercise the parallelism at all.

(I thought it does because it uses multiple deps in [some places](https://github.com/crystal-lang/shards/blob/635d36af83b3adfbfbc8af88e5f00a3c5af5d585/spec/integration/install_spec.cr#L344) -
but the fact that it didn't break for hg... confuses me 🤔)

Reply:
Hm. The best I could offer for now would be to _actually_ exclude hg from
parallelism (like I thought it already was), by doing something like:

```diff
          begin
-           dep.resolver.update_local_cache
+           dep.resolver.update_local_cache if dep.resolver.is_a? GitResolver
            ch.send(nil)
            active -= 1
          rescue ex : Exception
            ch.send(ex)
          end
        end
```

But I feel like if the spec's didn't catch the hg issue then they
might also not sufficiently cover the git-side. So even though it works
splendidly for me under real usage, spec-coverage should probably
be addressed before merging this.

If someone more familiar with the spec-suite and/or hg
has an idea on how to best do that, please don't hesitate. 😬

Reply:
> _actually_ exclude hg from parallelism

Fyi, this is now added here: https://github.com/crystal-lang/shards/pull/539/commits/886fdbd9a0e14104d201c234922f967bc5d593e9

Reply:
Agree, but that will need test-coverage first.
For git I have some confidence that it works fine since I've been using it daily for a while, but
I've never encountered a hg-shard in the wild, so don't even have an URL to test that side manually.

Not sure when I'll have time to continue on this, so excluding Hg for now.

Reply:
That's totally fine. Concurrent fetches are just an optimization without functional differences. There's no need to have this implemented for all resolvers immediately.

Reply:
Resolving this convo as I believe all points are addressed.

* Parallelism is restricted to only apply to GitResolver (because hg not implemented, yet)
* Flag has been renamed to `--jobs` to keep it generic (for consistency with `bundler`
  and for when we add hg in the future)
* Help-text still mentions git explicitly - but I believe that makes sense as to not
   confuse users into thinking it would apply to all resolvers.
   
Please re-open or reply on main thread if any of these need further discussion.

Reply:
I'd prefer to keep the wording more general, but mention that it only works for git currently. For example:

```
--jobs:: Number of repository downloads to perform in parallel (default: 8). Currently only functions for git resolver.
```

Reply:
Ok, updated in https://github.com/crystal-lang/shards/pull/539/commits/670cfb749d6f1890074a72891d25ae71679c07d6

Dropped "function" and "resolver" to make it shorter.